lib/passageidentity/error.rb

```diff
@@ -7,21 +7,45 @@ module Passage
   class PassageError < StandardError
     attr_reader :status_code, :error_code, :message, :status_text, :error
 
-    def initialize(message: nil, status_code: nil, body: nil)
-      super(message)
+    def initialize(message: nil, status_code: nil, body: {})
+      # Ensure body is a hash
+      body = {} if body.nil?
+      body = symbolize_keys(body)
```

Comment:
The addition here now supports bodies returned by the Passage Auth API, as well as errors filled in with SDK code like: https://github.com/passageidentity/passage-ruby/blob/b4ca0ae675975aa4ecef615214895f9968a059ff/lib/passageidentity/user_api.rb#L51-L57